src/main/java/net/dv8tion/jda/core/utils/WidgetUtil.java

```diff
@@ -191,35 +190,63 @@ public static Widget getWidget(String guildId) throws RateLimitedException
      */
     public static Widget getWidget(long guildId) throws RateLimitedException
     {
-        Args.notNull(guildId, "GuildId");
+        Checks.notNull(guildId, "GuildId");
+
+        HttpURLConnection connection;
+        int code;
+
         try
         {
-            HttpResponse<JsonNode> result = Unirest.get(String.format(WIDGET_URL, guildId)).asJson();
-            switch(result.getStatus())
-            {
-                case 200: return new Widget(result.getBody().getObject()); // ok
-                case 400:              // not valid snowflake
-                case 404: return null; // guild not found
-                case 403: return new Widget(guildId); // widget disabled
-                case 429: // ratelimited
+            connection = (HttpURLConnection) new URL(String.format(WIDGET_URL, guildId)).openConnection();
+            connection.setRequestMethod("GET");
+            connection.addRequestProperty("user-agent", Requester.USER_AGENT);
+            connection.connect();
```

Comment:
Why are we using HttpUrlConnection here instead of OkHttp?
Also, you need to add the `accept-encoding` header here just like in requester.